### DISCUSSION TITLE
Code Editor -> Folding behavior clarification needed

### DISCUSSION THREAD

**Discussion by seg729** [dbeaver/dbeaver#20640] [Q&A] [answered]
It seems like when _**creating a new script**_ , 'Foldings' is not enabled by default, even when checked in the editor's preferences.  

The script can be saved and renamed but it doesn't enable the 'Foldings' feature at that point either.  If the script is closed from the editor and reopened, then 'Foldings' is then enabled.  Is there a setting to enable Foldings from the start?  See screenshots and comments below for additional details.  

DBeaver: Community edition Version 23.1.2.
OS: MacOS

Thanks in advance!

<img width="724" alt="image" src="https://github.com/dbeaver/dbeaver/assets/140108645/b0aa7ea7-57ae-4b19-b731-0e66cbb2d9e2">

<img width="922" alt="image" src="https://github.com/dbeaver/dbeaver/assets/140108645/ee7c9c65-7113-4770-88f3-a411defb6a7b">

<img width="913" alt="image" src="https://github.com/dbeaver/dbeaver/assets/140108645/fa3dd67c-7083-4961-a338-7c3ade0efdca">

**Comment by E1izabeth** [comment, marked as answer]
Comments with -- are database-specific. You need to connect to your database first.
Common folding works always. For example, if you put where to the new line, the folding will appear for your select no matter you connected to the database or not.

**Reply by seg729** [reply]
Thank you
